### DISCUSSION TITLE
Add tool / function calling and tool messages to Prompt Management

### DISCUSSION THREAD

**Discussion by eloko7** [langfuse/langfuse#2624] [Ideas]
### Describe the feature or potential improvement

First of all thank you for your well designed project!
I would propose to add some additional fields to the prompt management of type `chat`.
When developing projects around tool calling and Agent workflows there is often a need for few-shot prompting with tool calling examples and the appropriate tool response message. Right now I have to use the config field and create the message object at runtime. Moreover, if observed that I can add a message with role "tool" in via Python SDK but not in the UI. 

**Requirements:**
 - Adding "Tool" Role to ChatMessage with additional argument "tool_call_id"
 - Adding "tool_calls" as an argument to assistant message
 - Adding "name" argument to all messages (to specify e.g. "example_assistant" & "example_user" according to OpenAI docs)

**Suggestion**
Since from current implementation there is no differentiation between arguments/fields and the message type I would be sufficient to stick with "role" and "content" and just add something like "additonal_kwargs" in the backend and the possibility to add more fields in the frontend.

Thanks.

**Comment by marcklingen** [comment]
New idea here: separate tool library that can be reused across prompts
https://github.com/orgs/langfuse/discussions/4674
